### ISSUE TITLE
Icons in components for a project with custom icons/fontawesome icons

### ISSUE THREAD

**Issue by WillianGruber** [stsrki/Blazorise#341]
Hello,

I just created a demo project using Material, but I won't be using Material Icons.

1) Testing the Sidebar component with an item and a subitem, the [expand_more] or [chevron_down] icon works.

2) Testing the Bar component, the BarDropdownToggle renders the name of the icon:
Code: <BarDropdownToggle >More</BarDropdownToggle>
Rendered: More expand_more
![image](https://user-images.githubusercontent.com/56033692/68704800-a8cf0300-056b-11ea-91bb-b86150765029.png)

3) Testing the Theme Selector component as in the demo page, the checkboxes are not rendered:
![image](https://user-images.githubusercontent.com/56033692/68704908-d7e57480-056b-11ea-8f4c-a324ccecf132.png)

Is it possible to customize the icons? Or adding a fallback? Or passing the as parameters (checkbox empty and checkbox filled)? Or in the component rendering, check if the default icons are registered, and if not, search for the alternative?

Thank you!

**Comment by stsrki** [comment]
If you don't want to use material icons you can implement your own implementation of `IIconProvider` and then register it with Blazorise instead of material icons.

The code for registration can be found [here](https://github.com/stsrki/Blazorise/blob/master/Source/Extensions/Blazorise.Icons.FontAwesome/Config.cs) where you can see how it can be done.

**Comment by WillianGruber** [comment]
But how can I use the FontAwesome Icons? They are already implemented and registered, and I use them in buttons, etc, so they work... but they won't work inside the Material components as they target the icon name in material icons...

**Comment by stsrki** [comment]
Sorry I misunderstood your question. Anyways, I guess you have already installed FontAwesome NuGet package? If not, you can find it [here](https://www.nuget.org/packages/Blazorise.Icons.FontAwesome/)

Next you need to [register](https://github.com/stsrki/Blazorise#4-registrations) the package with Blazorise and also add the [FontAwesome sources](https://github.com/stsrki/Blazorise#2-source-files)

After that you should be able to use the icons.

_NOTE:_ Currently there is only a small number of Icons available with the built-in enum `IconName`. So you should instead use the actual FontAwesome icon names eg. "fa-save".

See [#95](https://github.com/stsrki/Blazorise/issues/95)

**Comment by WillianGruber** [comment]
I probably expressed myself poorly, english is not my native language. Sorry.

What I mean is I have already done the steps you mentioned, and I can use the FA Icons that way in a button for example. I can also use the IconName enum. But inside components that use the icons themselves it does not work. I'll send some screenshots in half an hour when I get home.

**Comment by WillianGruber** [comment]
Here is an example using FA names and IconName enum:
`<Button Color="Color.Primary" Clicked="@Save"><Icon Name="@("fa-chevron-down")" Margin="Margin.Is3.FromRight" />Save</Button>`
`<Button Color="Color.Primary" Clicked="@Save"><Icon Name="IconName.ExpandMore" Margin="Margin.Is3.FromRight" />Save</Button>`
This renders to:
![image](https://user-images.githubusercontent.com/56033692/68723734-3aed0080-0598-11ea-9ae7-108d0c44935a.png)

But inside Blazorise components such as BarDropdownToggle the IconName.ExpandMore isn't interpreted correctly, as my 1st screenshot:
![image](https://user-images.githubusercontent.com/56033692/68723889-bea6ed00-0598-11ea-96d7-6e165a302159.png)

**Comment by stsrki** [comment]
Since you're seeing the icon named `expand_more` it means you have not registered the font-awesome icon provider in the `Startup.cs`. That's the only reason I can think of why this would happen.

You must do [this step](https://github.com/stsrki/Blazorise#4-registrations) and call `AddFontAwesomeIcons()`. Also call `UseFontAwesomeIcons()` in `Configure(...)` method.

**Comment by WillianGruber** [comment]
Here is my Startup.cs:
```csharp
public void ConfigureServices(IServiceCollection services)
        {
            services.AddRazorPages();
            services.AddServerSideBlazor();
            services.AddBlazorise(options => { options.ChangeTextOnKeyPress = true; }) // from v0.6.0-preview4
                .AddMaterialProviders()
                .AddFontAwesomeIcons();
        }

public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                app.UseExceptionHandler("/Error");
                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
                app.UseHsts();
            }

            app.UseHttpsRedirection();
            app.UseStaticFiles();

            app.UseRouting();

            app.ApplicationServices
                .UseMaterialProviders()
                .UseFontAwesomeIcons();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapBlazorHub();
                endpoints.MapFallbackToPage("/_Host");
            });
        }
```

Also my _Host.cshtml file contains:
```html
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Omniflow Blazor</title>
    <base href="~/" />

    <!-- Material CSS -->
    <link href="css/material.min.css" rel="stylesheet">

    <!-- Add Material font (Roboto) and Material icon as needed -->
    <link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,500,500i,700,700i|Roboto+Mono:300,400,700|Roboto+Slab:300,400,700" rel="stylesheet">

    <!-- FontAwesome Icons -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.4.1/css/all.css" integrity="sha384-5sAR7xN1Nv6T6+dT2mhtzEpVJvfS3NScPQTrOxhwjIuvcA67KV2R5Jz6kr4abQsz" crossorigin="anonymous">

    <!-- etc etc etc -->
</head>
```

That's how it was configured, and the steps you mentioned are there. Still, the problem persists.

**Comment by stsrki** [comment]
So, I tried your code and found the culprit. The problem is actually in [daemonite material](http://daemonite.github.io/material/)  where they're using a hard-coded icon names for some of the scss rules. 

**Problem:** https://github.com/Daemonite/material/blob/acad4df33b2f0da8fbc427a5bd45d4b7240e46d3/assets/scss/variables/_variable-bootstrap.scss#L36

It seems they're expecting to always use material icons with their library.

**Comment by WillianGruber** [comment]
I see. I don't mind using the Material Icons for the default components, mainly because it would need to change a lot in the css file.

I solved this by adding the nuget Blazorise.Icons.Material, updating _Host.cshtml with `<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">`, but **_NOT_** registering anything in Startup.cs. This way, I can use FA in anything I need, and the Blazorise components will use Material Icons.

While searching for a solution, I saw that FA5 has support for solid, regular and brand styles. I'll create another issue suggesting that implementation.

Thank you very much @stsrki
